src/main/app/form/validation.ts

```diff
@@ -95,7 +107,18 @@ export const isDateInputInvalid: DateValidator = date => {
   if (year === 0 && month === 0 && day === 0) {
     return;
   }
-  if (!dayjs(`${year}-${month}-${day}`, 'YYYY-M-D', true).isValid()) {
+
+  const monthLength = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+
+  if (year % 400 === 0 || (year % 100 !== 0 && year % 4 === 0)) {
+    monthLength[1] = 29;
+  }
+
+  if (month < 0 || month > 12) {
+    return invalid;
+  }
+
+  if (!(day >= 0 && day <= monthLength[month - 1])) {
```

Comment:
I think this block would be better if you used else ifs instead of multiple ifs.

Reply:
Fixed now